dali/operators/bbox/bbox_rotate.cc

```diff
@@ -0,0 +1,415 @@
+  }
+
+  // Clip to image coordinates and remove boxes (and labels) with too large a reduction.
+  const auto num_out_boxes =
+      ClipAndRemoveBoxes(out_boxes, remove_threshold, out_wh, in_labels, out_labels);
+
+  if (!ltrb || bbox_norm) {  // Convert back to xywh and/or normalized format if needed
+    out_boxes = dali::span(out_boxes.data(), num_out_boxes);  // handle if some boxes culled
+
+    // Recip of whwh to multiply with box for normalization
+    const vec4 box_norm_vec = {1.f / out_wh.x, 1.f / out_wh.y, 1.f / out_wh.x, 1.f / out_wh.y};
+
+    std::for_each(out_boxes.begin(), out_boxes.end(), [ltrb, bbox_norm, box_norm_vec](vec4& box) {
+      if (!ltrb) {
+        box.z -= box.x;
+        box.w -= box.y;
+      }
+      if (bbox_norm) {
+#pragma omp simd simdlen(4)
+        for (int i = 0; i < 4; ++i) {
+          box[i] *= box_norm_vec[i];
+        }
+      }
+    });
+  }
+
+  return num_out_boxes;
+}
+
+template <>
+void BBoxRotate<CPUBackend>::RunImpl(Workspace& ws) {
+  const auto& in_boxes = ws.Input<CPUBackend>(0);
+
+  auto& tpool = ws.GetThreadPool();
+  for (int sample_idx = 0; sample_idx < in_boxes.num_samples(); ++sample_idx) {
+    tpool.AddWork([&, sample_idx](int) {
+      std::optional<ConstSampleView<CPUBackend>> in_labels;
+      std::optional<SampleView<CPUBackend>> out_labels;
+      if (ws.NumInput() == 2) {
+        in_labels = ws.Input<CPUBackend>(1)[sample_idx];
+        out_labels = ws.Output<CPUBackend>(1)[sample_idx];
+      }
+
+      float angle;
+      if (spec_.HasTensorArgument("angle")) {
+        angle = ws.ArgumentInput("angle")[sample_idx].data<float>()[0];
+      } else {
+        angle = spec_.GetArgument<float>("angle");
+      }
+      angle = deg2rad(angle);
+
+      vec2 image_wh;
+      if (spec_.HasTensorArgument("input_shape")) {
+        auto shape_tensor = ws.ArgumentInput("input_shape")[sample_idx].data<std::int64_t>();
+        image_wh.x = shape_tensor[shape_wh_index_.first];
+        image_wh.y = shape_tensor[shape_wh_index_.second];
+      } else {
+        auto shape_tensor = this->spec_.GetRepeatedArgument<int>("input_shape");
+        image_wh.x = shape_tensor[shape_wh_index_.first];
+        image_wh.y = shape_tensor[shape_wh_index_.second];
+      }
+
+      vec2 canvas_wh;
+      if (keep_size_) {
+        canvas_wh = image_wh;
+        if (spec_.HasTensorArgument("size") || spec_.HasArgument("size")) {
+          DALI_FAIL("fn.bbox_rotate `keep_size` is mutually exclusive with `size` argument");
+        }
+      } else if (spec_.HasTensorArgument("size")) {
+        auto size_tensor = ws.ArgumentInput("size")[sample_idx].data<float>();
+        canvas_wh.x = size_tensor[1];
+        canvas_wh.y = size_tensor[0];
+      } else if (spec_.HasArgument("size")) {
+        auto size_tensor = this->spec_.GetRepeatedArgument<float>("size");
+        canvas_wh.x = size_tensor[1];
+        canvas_wh.y = size_tensor[0];
```

Comment:
These should be rounded the same way as in `rotate` and `warp_affine` (just cast to `int`).
Also, `warp_affine` and `rotate` will accept any type as the size, whereas here you require the tensor variant to be exactly `float`. See `warp_param_provider.h` `GetTypedPerSampleSize`.